src/Context/GtmContext.php

```diff
@@ -75,17 +76,58 @@ public function getDataLayerSettingShouldMatch($key, $regex) {
    * @return mixed
    * @throws \Exception
    */
-  protected function getDataLayerValue($key) {
-    $json_arr = $this->getDataLayerJson();
+    protected function getDataLayerValue($key)
+    {
+        $json_arr = $this->getDataLayerJson();
+        // Loop through the array and return the data layer value
+        foreach ($json_arr as $json_item) {
+            // Check if the key contains dot.
+            if (strpos($key, '.', 0) !== false) {
+                $value = $this->getDotValue($json_item, $key);
+                if (!is_null($value)) {
+                    return $value;
+                }
+            } elseif (isset($json_item[$key])) {
+                return $json_item[$key];
+            }
+        }
+        throw new \Exception($key . ' not found.');
+    }
 
-    // Loop through the array and return the data layer value
-    foreach ($json_arr as $json_item) {
-      if (isset($json_item[$key])) {
-        return $json_item[$key];
-      }
+    /**
+     * Convert dataLayer values to dot notation
+     */
+    private function getDotValue($arr, $key)
+    {
+        $dot = dot($arr);
+        // Check if the array contains dot.
+        if (strpos($key, '[', 0) !== false) {
+            // Trim before [
+            $pos = strpos($key, '[');
+            if ($pos !== false) {
+                $trimmedString = substr($key, 0, $pos);
+            }
+            $value = $dot->get($trimmedString);
+            if ($value !== NULL) {
+                // Filter the number from string.
+                $int_var = (int)filter_var($key, FILTER_SANITIZE_NUMBER_INT);
```

Comment:
This is bad code. If the keys contains numbers, it will cause problems.
Please use regex to find the key by extracting numbers between the brackets `[1]`